src/testdir/test_quickfix.vim

```diff
@@ -6718,4 +6718,24 @@ func Test_hardlink_fname()
   call Xtest_hardlink_fname('l')
 endfunc
 
+func Test_event_QufickFixCmdChanged()
+  let g:count = 0
+  autocmd QuickFixCmdChanged * let g:count = g:count + 1
+
+  let g:l = [{'text': 'foo'}] + repeat([{'text': join(range(30))}], 2)
+  copen
+  call feedkeys(":call setqflist(g:l, 'r')\<CR>", "tx")
```

Comment:
This test only uses quickfix list.  Can you add tests for the location list also?

Reply:
I think there in include loclist when g:count is 3.. but the current test cases are somewhat simple. More extensive operations are needed.